components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -78,6 +88,10 @@ public Action addAction(String actionType, Action action, String tenantDomain) t
         validateMaxActionsPerType(resolvedActionType, tenantDomain);
         doPreAddActionValidations(action);
         String generatedActionId = UUID.randomUUID().toString();
+        if (LoggerUtils.isEnableV2AuditLogs()) {
+            JSONObject data = auditLogger.generateAuditLogEntry(ADD_ACTION, actionType, null, action);
+            auditLogger.printAuditLog(ADD_ACTION, data);
+        }
         return CACHE_BACKED_DAO.addAction(resolvedActionType, generatedActionId, action,
```

Comment:
So the real update happens here. That means audit log should be done after this

Reply:
Updated with a105238b1c38f7405ecf43ecaa9c07b9b0de7b7f